### PR TITLE
Fix lstm backward arguments

### LINKED ISSUE #4319: Can't use chainer.grad for LSTM
I tried to differentiate a LSTM block with `chainer.grad` function along with `enable_double_backprop=True` option. Then got the error:

Code:
```
import numpy as np
import chainer
import chainer.links as L

x = chainer.Variable(np.random.rand(10, 20).astype('f'))
lstm = L.LSTM(20, 20)
y = lstm(x)
dydx, = chainer.grad([y], [x], enable_double_backprop=True)
```

Error:
```
~/.pyenv/versions/anaconda3-5.0.0/lib/python3.6/site-packages/chainer/functions/activation/lstm.py in backward(self, indexes, grads)
    111     def backward(self, indexes, grads):
    112         grad_inputs = (
--> 113             self.get_retained_inputs() + self.get_retained_outputs() + grads)
    114         return LSTMGrad()(*grad_inputs)
```

Is this a bug of LSTM module or my wrong usage of LSTM?
I'm using Chainer v4.0.0b3.

Any comments will help. Thanks.

### PR DESCRIPTION
**Pull request by mitmul**
This PR partially resolves #4319

### CODE REVIEW

**Comment by toslunar** [comment]
#4186 would fix this.

**Comment by okuta** [comment]
Cloud you add test?

File: chainer/functions/activation/lstm.py [hunk 1]
```diff
@@ -109,6 +109,8 @@ def forward(self, inputs):
         return c_next, h
 
     def backward(self, indexes, grads):
+        if not isinstance(grads, tuple):
+            grads = tuple(grads)
```
Comment (niboshi) [inline chainer/functions/activation/lstm.py]: This is not needed (once #4186 is merged).

**Comment by okuta** [comment]
LGTM!

**Comment by okuta** [comment]
Please add a test as another PR.

jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Pull request merged**